pages/magic.tsx

```diff
@@ -0,0 +1,47 @@
+import { createSessionCookie } from '@lib/helpers/backend/session-cookie';
+import { consumeMagicLink } from '@lib/services/magic';
+import { StatusCodes } from 'http-status-codes';
+import { GetServerSideProps, NextPage } from 'next';
+
+interface MagicProps {
+  message: string;
+}
+
+const Magic: NextPage<MagicProps> = ({ message }) => {
+  return <p>{message}</p>;
+};
+
+export const getServerSideProps: GetServerSideProps = async ({ res, query }) => {
+  const { token } = query;
+
+  if (typeof token !== 'string') {
+    res.statusCode = StatusCodes.BAD_REQUEST;
+    return {
+      props: {
+        message: 'Invalid URL',
```

Comment:
I think this message could be a bit more descriptive, maybe something like "invalid login link"